### DISCUSSION TITLE
airflow showing hearbeat issue

### DISCUSSION THREAD

**Discussion by Kathan1210** [apache/airflow#23139] [General]
My airflow version 2.2.2 shows that "The Scheduler does not appear to be running. The last heartbeat was received %d minutes ago". I want help about why the issue is occurring and how to mitigate it. This issue is not frequent but it occurs sometimes and currently when I restart the scheduler once again. This issue is getting resolved but this is not how it should be resolved.

The scheduler logs show that there is some deadlock which is happening during this time but I am not able to figure out the why the deadlock is happening.

{scheduler_job.py:1114} INFO - Resetting orphaned tasks for active dag runs
{scheduler_job.py:644} ERROR - Exception when executing SchedulerJob._run_scheduler_loop
Traceback (most recent call last):
File "/usr/local/lib64/python3.6/site-packages/mysql/connector/connection_cext.py", line 519, in cmd_query
query_attrs=self._query_attrs)
_mysql_connector.MySQLInterfaceError: Deadlock found when trying to get lock; try restarting transaction

Also the current parameters for airflow cfg scheduler are as below
job_heartbeat_sec = 5
scheduler_heartbeat_sec = 5
num_runs = -1
scheduler_idle_sleep_time = 1
min_file_process_interval = 300
dag_dir_list_interval = 300
print_stats_interval = 30
pool_metrics_interval = 5.0
scheduler_health_check_threshold = 30
scheduler_zombie_task_threshold = 300
catchup_by_default = True
max_tis_per_query = 512
max_dagruns_to_create_per_loop = 10
orphaned_tasks_check_interval = 300.0

Please help if anyone is aware about how to solve this issue?

**Comment by potiuk** [comment]
I think you should try to provide more information and open an issue with logs - especially around the deadlock time when it happens. That will allow to make a deeper investigation. You should provide not only the client logs but also MySQL server logs from around the deadlock, because this is the only way to find what the client query deadlocsk with (only server can print such information)

You can look through a number of past issues in Airflow with "deadlock" and see examples of logs people provided/
